File: fips-files/verbs/vic20-via-tests.py
import os

from mod import log, util, project, config

TestDir = '/tests/vice-tests/VIC20/'

Tests = [
    'viavarious/via1.prg',
    'viavarious/via2.prg',
    'viavarious/via3.prg',
    'viavarious/via3a.prg',
    'viavarious/via4.prg',
    'viavarious/via4a.prg',
    'viavarious/via5.prg',
    'viavarious/via5a.prg',
    'viavarious/via9.prg',
    'viavarious/via10.prg',
    'viavarious/via11.prg',
    'viavarious/via12.prg',
    'viavarious/via13.prg',
]

#-------------------------------------------------------------------------------
def match_test(test_name, whitelist):
    if not whitelist:
        return True
    else:
        for substr in whitelist:
            if substr in test_name:
                return True
    return False

#-------------------------------------------------------------------------------
def run(fips_dir, proj_dir, args) :
    cfg = config.get_default_config();
    filt = args
    for test in Tests:
        if match_test(test, args):
            path = proj_dir + TestDir + test
            print("> running test '{}':".format(test))
            project.run(fips_dir, proj_dir, cfg, 'vic20-ui', ['exp=ram8k', 'file={}'.format(path)], None)

#-------------------------------------------------------------------------------
def help() :
    log.info(log.YELLOW +
        'fips vic20-via-tests\n' +
        'fips vic20-via-tests via1 via2 ...' +
        log.DEF +
        '    run all or selected VIC-20 VIA tests from the VICE test bench')

File: fibs-scripts/vic20-via-tests.ts
// helper command to run VIC20 VIA tests
import { Configurer, Project, log } from 'jsr:@floooh/fibs@^1';

export function addVic20TestCommand(c: Configurer) {
    c.addCommand({ name: 'vic20-via-tests', help, run });
}

function help() {
    log.helpCmd([
        'vic20-via-tests',
        'vic20-via-tests [via1 via2 ...]',
    ], 'run vic20 via tests');
}

async function run(p: Project, args: string[]): Promise<void> {
    const testDir = 'tests/vice-tests/VIC20/'
    const allTests = [
        'viavarious/via1.prg',
        'viavarious/via2.prg',
        'viavarious/via3.prg',
        'viavarious/via3a.prg',
        'viavarious/via4.prg',
        'viavarious/via4a.prg',
        'viavarious/via5.prg',
        'viavarious/via5a.prg',
        'viavarious/via9.prg',
        'viavarious/via10.prg',
        'viavarious/via11.prg',
        'viavarious/via12.prg',
        'viavarious/via13.prg',
    ];
    let tests;
    if (args.length === 1) {
        tests = allTests;
    } else {
        tests = allTests.filter((t) => {
            const filename = t.match(/viavarious\/(.+)\.prg$/)?.[1];
            return filename && args.slice(1).some((a) => filename === a);
        });
    }
    for (const test of tests) {
        const c = p.activeConfig();
        const path = `${p.dir()}/${testDir}/${test}`;
        await c.runner.run(p, c, p.target('vic20-ui'), { args: ['exp=ram8k', `file=${path}`] });
    }
}